### PR TITLE
Add information on @Mod

### PR DESCRIPTION
**Pull request by under-scores**
As it has been almost a year and a half since the last update to #46, I thought someone should probably fill in the information for `@Mod`.

### CODE REVIEW

File: docs/gettingstarted/structuring.md [hunk 1]
```diff
@@ -67,7 +67,29 @@ and will contain some special indicators marking it as such.
...
+|                      useMetadata |       boolean      |      false     | If set to true, properties in `@Mod` will be overridden by `mcmod.info`. |
+| clientSideOnly<br>serverSideOnly | boolean<br>boolean | false<br>false | If either is set to `true`, the mod will only run on the respective side. If both are true, the game crashes. |
+|        acceptedMinecraftVersions |       String       |       ""       | The version range of Minecraft the mod will run on.[\*](#version-ranges) An empty string will match all versions. |
+|         acceptableRemoteVersions |       String       |       ""       | Specifies a remote version range that this mod will accept as valid.[\*](#version-ranges) `""` Matches the current version, and `"*"` matches all versions. |
+|           acceptableSaveVersions |       String       |       ""       | A version range specifying compatible save version information.[\*](#version-ranges) If you follow an unusual version convention, use `SaveInspectionHandler` instead. |
```
Comment (under-scores) [inline docs/gettingstarted/structuring.md]: Tbh, I'm not exactly positive what this means :P
Is it referring to which world saves are compatible with each version of the mod?

Comment (under-scores) [inline docs/gettingstarted/structuring.md]: Tbh, I'm not exactly positive what this means :P
Is it referring to which world saves are compatible with each version of the mod?

File: docs/gettingstarted/structuring.md [hunk 2]
```diff
@@ -67,7 +67,29 @@ and will contain some special indicators marking it as such.
...
+This is an annotation indicating to the Forge Mod Loader that the class is a Mod entry point. It contains various metadata about the mod. It also designates the class that will receive `@EventHandler` events.
+
+Here is a table of the properties of `@Mod`:
+
+|                         Property |        Type        |    Default     | Description |
+|---------------------------------:|:------------------:|:--------------:|:------------|
+|                            modid |       String       |    required    | A unique identifier for the mod. It must be lowercased, and will be truncated to 64 characters in length. |
+|                             name |       String       |       ""       | A user-friendly name for the mod. |
+|                          version |       String       |       ""       | The version of the mod. It should be just numbers seperated by dots. Even if `useMetadata` is set to `true`, it's a good idea to put the version here anyways. |
```
Comment (tterrag1098) [inline docs/gettingstarted/structuring.md]: Please mention SemVer

File: docs/gettingstarted/structuring.md [hunk 3]
```diff
@@ -67,7 +67,29 @@ and will contain some special indicators marking it as such.
...
+|                     dependencies |       String       |       ""       | Dependencies for the mod. The specification is described in the Forge `@Mod` javadoc:<br><blockquote><p>A dependency string can start with the following four prefixes: `"before"`, `"after"`, `"required-before"`, `"required-after"`; then `":"` and the `modid`.</p><p>Optionally, a version range can be specified for the mod by adding `"@"` and then the version range.[\*](#version-ranges)</p><p>If a "required" mod is missing, or a mod exists with a version outside the specified range, the game will not start and an error screen will tell the player which versions are required.</p>
+|                      useMetadata |       boolean      |      false     | If set to true, properties in `@Mod` will be overridden by `mcmod.info`. |
+| clientSideOnly<br>serverSideOnly | boolean<br>boolean | false<br>false | If either is set to `true`, the mod will only run on the respective side. If both are true, the game crashes. |
```
Comment (tterrag1098) [inline docs/gettingstarted/structuring.md]: It's less that the mod "will only run" and more that the jar is *entirely* skipped

File: docs/gettingstarted/structuring.md [hunk 4]
```diff
@@ -67,7 +67,29 @@ and will contain some special indicators marking it as such.
...
+|           acceptableSaveVersions |       String       |       ""       | A version range specifying compatible save version information.[\*](#version-ranges) If you follow an unusual version convention, use `SaveInspectionHandler` instead. |
+|           certificateFingerprint |       String       |       ""       | See the tutorial on [jar signing](../concepts/jarsigning.md). |
+|                      modLanguage |       String       |     "java"     | The programming language the mod is written in. Can be either `"java"` or `"scala"`. |
+|               modLanguageAdapter |       String       |       ""       | Path to a language adapter for the mod. The class must have a default constructor and must implement `ILanguageAdapter`. If it doesn't, Forge will crash. If set, overrides `modLanguage`. |
+|                 canBeDeactivated |       boolean      |      false     | If the mod can be deactivated (e.g. a minimap mod), set this to `true` and [recieve](../events/intro.md#creating-an-event-handler) `FMLDeactivationEvent` to perform cleanup tasks. |
```
Comment (tterrag1098) [inline docs/gettingstarted/structuring.md]: This is not implemented

File: docs/gettingstarted/structuring.md [hunk 5]
```diff
@@ -67,7 +67,29 @@ and will contain some special indicators marking it as such.
...
+|                       guiFactory |       String       |       ""       | Path to the mod's gui factory, if one exists. It must implement `IModGuiFactory`. |
```
Comment (tterrag1098) [inline docs/gettingstarted/structuring.md]: Maybe some more explanation on what this is for?

File: docs/gettingstarted/structuring.md [hunk 6]
```diff
@@ -75,18 +75,18 @@ Here is a table of the properties of `@Mod`:
...
 |                      modLanguage |       String       |     "java"     | The programming language the mod is written in. Can be either `"java"` or `"scala"`. |
 |               modLanguageAdapter |       String       |       ""       | Path to a language adapter for the mod. The class must have a default constructor and must implement `ILanguageAdapter`. If it doesn't, Forge will crash. If set, overrides `modLanguage`. |
-|                 canBeDeactivated |       boolean      |      false     | If the mod can be deactivated (e.g. a minimap mod), set this to `true` and [recieve](../events/intro.md#creating-an-event-handler) `FMLDeactivationEvent` to perform cleanup tasks. |
-|                       guiFactory |       String       |       ""       | Path to the mod's gui factory, if one exists. It must implement `IModGuiFactory`. |
+|                 canBeDeactivated |       boolean      |      false     | This is not implemented, but if the mod could be deactivated (e.g. a minimap mod), this would be set to `true` and the mod would [recieve](../events/intro.md#creating-an-event-handler) `FMLDeactivationEvent` to perform cleanup tasks. |
```
Comment (PaleoCrafter) [inline docs/gettingstarted/structuring.md]: Small typo here, it's `receive`, not `recieve` ^^

File: docs/gettingstarted/structuring.md [hunk 7]
```diff
@@ -75,18 +75,18 @@ Here is a table of the properties of `@Mod`:
...
 |         acceptableRemoteVersions |       String       |       ""       | Specifies a remote version range that this mod will accept as valid.[\*](#version-ranges) `""` Matches the current version, and `"*"` matches all versions. |
 |           acceptableSaveVersions |       String       |       ""       | A version range specifying compatible save version information.[\*](#version-ranges) If you follow an unusual version convention, use `SaveInspectionHandler` instead. |
 |           certificateFingerprint |       String       |       ""       | See the tutorial on [jar signing](../concepts/jarsigning.md). |
 |                      modLanguage |       String       |     "java"     | The programming language the mod is written in. Can be either `"java"` or `"scala"`. |
 |               modLanguageAdapter |       String       |       ""       | Path to a language adapter for the mod. The class must have a default constructor and must implement `ILanguageAdapter`. If it doesn't, Forge will crash. If set, overrides `modLanguage`. |
```
Comment (PaleoCrafter) [inline docs/gettingstarted/structuring.md]: I'd wait for MinecraftForge/MinecraftForge#4029 to be done and merged before this does, as the docs should be relevant for the latest version and lang adapters are subject to change.

**Comment by tterrag1098** [comment]
We'll add changes for that PR when necessary, for now this is good info to have.

**Pull request merged**